tsdb/querier.go

```diff
@@ -872,22 +889,14 @@ func (s *chunkSeries) Labels() labels.Labels {
 	return s.labels
 }
 
-func (s *chunkSeries) Iterator() SeriesIterator {
+func (s *chunkSeries) Iterator() chunkenc.Iterator {
 	return newChunkSeriesIterator(s.chunks, s.intervals, s.mint, s.maxt)
 }
 
-// SeriesIterator iterates over the data of a time series.
-type SeriesIterator interface {
-	// Seek advances the iterator forward to the given timestamp.
-	// If there's no value exactly at t, it advances to the first value
-	// after t.
-	Seek(t int64) bool
-	// At returns the current timestamp/value pair.
-	At() (t int64, v float64)
-	// Next advances the iterator by one.
-	Next() bool
-	// Err returns the current error.
-	Err() error
+func (s *chunkSeries) ChunkIterator() ChunkIterator {
+	// TODO(bwplotka): This is wrong. We probably have to be strict in terms of min/maxt and tombstones.
```

Comment:
TODO/TO decide:

When we want to apply tombstones in `ChunkIterator`? :/ 

TBH it might be the most efficient to pass intervals upstream e.g in streamed remote read protocol as well (???)

cc @brian-brazil

Reply:
Two options:

A) Naive: Rewrite chunk on the fly applying `mint,maxt` and `dranges`. It is must have for `Select` interface which assumes things are scoped to given time range and tombstones are tolerated.
B) Ignore mint,maxt level. Maybe allow ChunkIterator to return `deleted` intervals as well. This probably means extending the remote read chunked protocol [here](https://github.com/prometheus/prometheus/pull/5703) for passing tombstones intervals as well -> let client side to apply intervals.. However, we leak "deleted" metrics. If they were deleted because of GDPR or something, this is kind of wrong (: 

I think overall we need to do A here. Thoughts @brian-brazil @tomwilkie @codesome ?

Reply:
Definitely A, for general data security we don't want to be returning deleted data. It'd also push that complexity onto all readers, and they'd probably forget it.

Reply:
>  It'd also push that complexity onto all readers, and they'd probably forget it.

You would NOT put it to readers you mean? 

Makes sense to me.. Let's do it then here.